daffodil-tdml-processor/src/main/scala/org/apache/daffodil/tdml/SchemaCache.scala

```diff
@@ -33,6 +33,21 @@ object ENoWarnTDML { EqualitySuppressUnusedImportWarning() }
  */
 object SchemaDataProcessorCache extends SchemaCache[(Seq[Diagnostic], DFDL.DataProcessor), Seq[Diagnostic]]
 
+object SchemaCache {
+  /**
+   * A key object for caching compiled DFDL schemas. Everything that effects compilation
+   * must be captured here.
+   */
+  case class Key (
+    uri: URISchemaSource,      // The DFDL schema's URI
+    useSerializedProcessor: Boolean,
+    compileAllTopLevels: Boolean,
+    optRootName: Option[String],
+    optRootNamespace: Option[String],
+    tunables: Map[String, String])  // tunables map - required since tunables effect compilation
+  // (ex: parseUnparsePolicy, unqualifiedPathStepPolicy, perhaps others)
+
+}
```

Comment:
This is part of fixing a bug where we needed the tunables in the SchemaCache key or they would return a cached result incorrectly when tunables had changed the compilation parameters around namespaces. That got aggravated as part of debugging SAX tests with the various namespace calls. 

It's an interesting point though. If as part of debugging a problem you fix things and improve them that turn out to have been unrelated to your orignal problem, do you back out those changes? Often it's not easy to identify what goes with what, and one could spend a lot of time backing things out and breaking stuff to isolate the changes. 

I cannot tell you if, in conjunction with the other changes, this change to the schema cache would still cause a failure. But it was not correct before because the tunables weren't part of the key. So this is more correct, albeit possibly unrelated. I really just do not know any more if this fix would have mattered.